### PR TITLE
arcticons-sans: init at 0.580

### PR DESCRIPTION
**Pull request by asininemonkey**
## Description of changes

Add Arcticons Sans font: https://github.com/Arcticons-Team/Arcticons-Font

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/data/fonts/arcticons-sans/default.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  pname = "arcticons-sans";
+
+  src = fetchzip {
+    hash = "sha256-BRyYHOuz7zxD1zD4L4DmI9dFhGePmGFDqYmS0DIbvi8=";
+    url = "https://github.com/arcticons-team/arcticons-font/archive/refs/tags/${finalAttrs.version}.zip";
+  };
+
+  version = "0.580";
+})
```
Comment (Janik-Haag) [inline pkgs/data/fonts/arcticons-sans/default.nix:29]:
```suggestion
stdenvNoCC.mkDerivation (finalAttrs: {
  pname = "arcticons-sans";
  version = "0.580";

  src = fetchzip {
    hash = "sha256-BRyYHOuz7zxD1zD4L4DmI9dFhGePmGFDqYmS0DIbvi8=";
    url = "https://github.com/arcticons-team/arcticons-font/archive/refs/tags/${finalAttrs.version}.zip";
  };

  installPhase = ''
    runHook preInstall

    install -D -m444 -t $out/share/fonts/opentype $src/*.otf

    runHook postInstall
  '';

  meta = {
    description = "Arcticons Sans";
    homepage = "https://github.com/arcticons-team/arcticons-font";
    license = lib.licenses.ofl;
    maintainers = with lib.maintainers; [ asininemonkey ];
  };
})
```
Please sort the attributes in the correct order.
The long description is intended to be loonnnggg so multiple sentences, so I think we can just drop it here. 
Setting platforms to all is unneeded since the implicit platforms attribute set by stdenvNoCC should be fine.

File: pkgs/data/fonts/arcticons-sans/default.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
+{ fetchzip, lib, stdenvNoCC }:
```
Comment (Janik-Haag) [inline pkgs/data/fonts/arcticons-sans/default.nix:1]:
```suggestion
{ lib
, stdenvNoCC 
, fetchzip
}:
```
small formatting thing

**Review by Janik-Haag — COMMENTED** [review]
sorry that it took me so long to get back to you.

**Comment by asininemonkey** [comment]
> sorry that it took me so long to get back to you.

Not a problem @Janik-Haag and thank you for the feedback. I've made the requested changes and tidied up my commits.

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -1743,6 +1743,12 @@
     githubId = 113123021;
     name = "Ashvith Shetty";
   };
+  asininemonkey = {
+    email = "nixpkgs@asininemonkey.com";
+    github = "asininemonkey";
+    githubId = 65740649;
+    name = "Jose Cardoso";
+  };
```
Comment (Janik-Haag) [inline maintainers/maintainer-list.nix:1751]: this should be its own commit called: `maintainers: add asininemonkey`

**Review by Janik-Haag — COMMENTED** [review]
Eval fails because the `arcticons-sans` package isn't included into a package set. You have two options here, either add a entry to `pkgs/top-level/all-packages.nix` or move the package into `pkgs/by-name/ar/arctions-sans` and rename default.nix to package.nix then it will be picked up automatically.

File: maintainers/maintainer-list.nix (hunk 3 above)
Comment (asininemonkey) [reply maintainers/maintainer-list.nix:1751]: I've broken this out as requested.

**Comment by asininemonkey** [comment]
@Janik-Haag I've my original commit to `pkgs/by-name/ar/arctions-sans/package.nix` as suggested.

**Review by Janik-Haag — APPROVED** [review]
LGTM
